ComplianceAsCode/compliance-operator
# Add unique profile ID

This commit implements unique profile ID feature, we are adding a unique profile ID to Profile, ComplianceScan, and ComplianceCheckResult CRD. The profile UUID is generated from sha1 of `<product-id>-<existing-xccdf-profile-id>`

ex. `ocp4-moderate` profile has `redhat_openshift_container_platform_4.1` product id and `xccdf_org.ssgproject.content_profile_moderate` profile id

QA: 
What is the reason for this commit?
```
Provides a unique GID for each profile across all profile bundles/datastream,
so ACS can consume that to filter rules/compliancecheckresult by profile-id.
```
What is the business value?
```
Supporting integration with ACS
```
What would be the impact on the user once this goes live?
```
No impact should be expected with PR, user will see additional annotations/labels on Profile, ComplianceScan, TailoredProfile CRs.
Added annotations, labels:
"compliance.openshift.io/guid"
"compliance.openshift.io/extended-profile-guid"
it will be internal use for ACS
```
Would this apply to only new deployments or all the deployments once upgraded?
```
No impact
```
Would each user have different profile ID? Why is it good/necessary?
```
No, they will have the same Profile ID
```
 https://issues.redhat.com/browse/CMP-2452
